### PR TITLE
Added support for the base function

### PR DESCRIPTION
**Pull request by paulkned**
This is:

```
- [ ] a bugfix
- [ x ] a new feature
```

Checklist:

- [ x ] Changes are covered by unit tests
- [ x ] Code style is respected
- [ x ] Commit message explains **why** the change is made (see https://github.com/erlang/otp/wiki/Writing-good-commit-messages)
- [ x ] CHANGELOG.md contains a short summary of the change
- [ ] Documentation is updated as necessary

### Why this change is needed?

Support for the BASE function was missing, this pull requests adds it.

### CODE REVIEW

File: src/PhpSpreadsheet/Calculation/MathTrig.php [hunk 1]
```diff
@@ -84,6 +84,50 @@ public static function ATAN2($xCoordinate = null, $yCoordinate = null)
...
+            if ($number < 0 || $number >= 2 ** 53 || $radix < 2 || $radix > 36) {
+                return Functions::NAN(); // Numeric range constraints
+            }
+
+            $outcome = (string) base_convert($number, 10, $radix);
+            if ($minLength !== null) {
+                $outcome = str_pad($outcome, $minLength, '0', STR_PAD_LEFT); // String padding
+            }
+
+            return $outcome;
```
Comment (MarkBaker) [inline src/PhpSpreadsheet/Calculation/MathTrig.php]: Subtle difference: PHP's base_convert() will return lower-case alpha characters; MS Excel BASE returns upper-case alpha

File: src/PhpSpreadsheet/Calculation/Calculation.php [hunk 2]
```diff
@@ -328,6 +328,11 @@ class Calculation
             'functionCall' => [Functions::class, 'DUMMY'],
             'argumentCount' => '1',
         ],
+        'BASE' => [
+            'category' => Category::CATEGORY_MATH_AND_TRIG,
+            'functionCall' => [MathTrig::class, 'BASE'],
+            'argumentCount' => '3',
```
[the remark this answers is not in the document]
Comment (paulkned) [reply src/PhpSpreadsheet/Calculation/Calculation.php]: Fixed

File: src/PhpSpreadsheet/Calculation/MathTrig.php [hunk 3]
```diff
@@ -84,6 +84,50 @@ public static function ATAN2($xCoordinate = null, $yCoordinate = null)
...
+     *
+     * Excel Function:
+     *        BASE(Number, Radix [Min_length])
+     *
+     * @category Mathematical and Trigonometric Functions
+     *
+     * @param float $number
+     * @param float $adix
+     * @param int $minLength
+     * @param mixed $radix
```
[the remark this answers is not in the document]
Comment (paulkned) [reply src/PhpSpreadsheet/Calculation/MathTrig.php]: Yeah, sorry for that. Fixed.

File: src/PhpSpreadsheet/Calculation/MathTrig.php (hunk 1 above)
Comment (paulkned) [reply src/PhpSpreadsheet/Calculation/MathTrig.php]: Added a strtoupper.

**Comment by paulkned** [comment]
Thanks for the review. I fixed the (sloppy) problems and added an additional test for a radix of 16.

**Pull request merged**